### DISCUSSION TITLE
migrating existing ceph clusters to rook

### DISCUSSION THREAD

**Discussion by yeoldegrove** [rook/rook#13792] [Ideas]
I want to revive the question from https://github.com/rook/rook/discussions/12045 if migrations of existing ceph clusters to rook could be supported.

We might have the use case inside the [Sovereign Cloud Stack Community](https://scs.community) to migrate clusters deployed via [ceph-ansible](https://github.com/ceph/ceph-ansible) to be fully managed by rook.                         
                                                                                                                     
[Rook External Storage Cluster](https://rook.io/docs/rook/v1.12/CRDs/Cluster/external-cluster/) is not what we want, because we want to get rid of the old deployment and management solution `ceph-ansible`.
                                                                       
Some ground work was already done by us internally to answer the question if a migration would be possible at all.
We actually managed already to bootstrap a rook cluster (via applying a custom configmap `rook-ceph-mon-endpoints`) that joins an existing "old cluster" add later swap out MONs/MGRs  from the "old cluster" and also imported OSDs in rook.

At this point of time we're even confident that this could be a "near online migration experience".           
                                                                                                                   
We just want to know if there would be an interest to somehow document or even support this in the rook context. We also might have some resources to do that initially.                                                         
                                                                           
What also would be interesting is if there already is a tool that genereated rook-CRDs based on an existing ceph cluster?                                                                                                                
                                                                           
Objections and suggestions to additonal challenges are also welcome.

**Comment by parth-gr** [comment]
Hey @yeoldegrove ,
If you wanna migrate the ceph cluster to rook here is an external document you can look at https://docs.starlingx.io/operations/ceph_cluster_migration.html.

**Reply by yeoldegrove** [reply]
We actually looked at that and found it very useful, yes :+1: .
But we want an officially supported or at least "somewhat affiliated with rook" solution.

**Reply by parth-gr** [reply]
we don't have something with rook docs because of `various types of deployments`, you can try this one, if it is not helpful you can propose a new feature in rook issues.

**Reply by travisn** [reply]
@yeoldegrove The reality is Rook maintainers don't ever run non-Rook clusters, so it will not be realistic for us to try to support the migration of non-rook clusters to rook clusters. Even if it's just documentation, we wouldn't be able to validate it. The most feasible way to get it into the docs is if multiple community members can collaborate on it. Or you might write a blog about your experience with the migration, then we could link to it from this discussion or even from the Rook docs.

**Reply by brueggemann** [reply]
Just to clear that up, we have to write some tool to migrate ceph-ansible setups to rook, as we need that in the Sovereign Cloud Stack project.
We had a meeting about this topic and there the question came up, if we should do this by officially contributing to the rook project or by writing an external tool. So if migration is not a feature that is in scope by you, we are completely okay to write an external tool and publish that to the community via the Souvereign Cloud Stack repos.

**Reply by travisn** [reply]
The external tool sounds good then. Happy to answer any questions to help you through the exercise if you hit any snags.
